core_lib/src/managers/backupmanager.cpp

```diff
@@ -0,0 +1,678 @@
+/*
+
+Pencil2D - Traditional Animation Software
+Copyright (C) 2005-2007 Patrick Corrieri & Pascal Naidon
+Copyright (C) 2008-2009 Mj Mendoza IV
+Copyright (C) 2012-2020 Matthew Chiawen Chang
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; version 2 of the License.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+*/
+
+#include "object.h"
+#include "editor.h"
+
+#include <QAction>
+#include <QDebug>
+#include <QSettings>
+
+
+#include "preferencemanager.h"
+#include "layermanager.h"
+#include "soundmanager.h"
+#include "backupmanager.h"
+#include "selectionmanager.h"
+
+#include "backupelement.h"
+#include "legacybackupelement.h"
+
+#include "layerbitmap.h"
+#include "layervector.h"
+#include "layersound.h"
+
+
+#include "bitmapimage.h"
+#include "vectorimage.h"
+#include "soundclip.h"
+
+BackupManager::BackupManager(Editor* editor) : BaseManager(editor, "BackupManager")
+{
+    qDebug() << "BackupManager: created";
+}
+
+BackupManager::~BackupManager()
+{
+    if (!mNewBackupSystemEnabled)
+    {
+        clearStack();
+    }
+    delete mUndoStack;
```

Comment:
This "delete" is not necessary (although ok) since BackupManager has a QObject parent.